src/ViewerComponents/ImageViewer.svelte

```diff
@@ -0,0 +1,120 @@
+<script>
+    import {onMount} from 'svelte';
+    import {token} from "../store";
+    import OpenSeadragon from 'openseadragon';
+
+    export let resource, ontology, server, user, shortname, shortcode;
+    let error = false;
+    let container
+    let viewer
+
+    onMount(() => {
+        container.id = Math.random().toString(36).substring(8);
+    })
+
+    /**
+     * Logs in to the provided server and saves the token into the store.
+     *
+     * @returns {Promise<void>}
+     */
+    async function login() {
+        error = false;
+
+        const res = await fetch(`https://${server}/v2/authentication`,
+            {
+                headers: new Headers({
+                    'Content-Type': 'application/json'
+                }),
+                method: 'POST',
+                body: JSON.stringify({'email': user['Email'], 'password': user['Pwd']})
+            })
+
+        // Checks if request succeeded
+        if (!res.ok) {
+            error = true;
+            console.error(res);
+            return;
+        }
+
+        const json = await res.json();
+        token.set(json.token);
+
+        await resourceRequest();
```

Comment:
Doesnt seem the right place to load the resource

Reply:
I will extract this and move in to dsp services later

Reply:
-> dsp services